src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -147,9 +145,13 @@ mineNakamoto probed difiFn getBlocks unsealedBlock = do
   where
     mine :: BlockHeader c Unsealed -> PoW -> Maybe PoW
     mine hdr (PoW nonce)
-        | hasPoW hdr        = Just (PoW nonce)
-        | nonce < maxBound  = mine hdr (PoW (nonce + 1))
-        | otherwise         = Nothing
+        | hasPoW (sealHeader (PoW nonce) hdr) = Just (PoW nonce)
```

Comment:
can we use `pow@(PoW nonce)` to not repeat this here?

Reply:
maybe, I can try 😉

Reply:
386578cc